integr8ly/integreatly-operator
# [INTLY-4308] Ensure cro resources are deleted before removing namespace finalizer

### Description 
Currently the finalizer logic for the cloud resources product simply deletes the namespace in which it's installed.This causes the cloud resource operator to shut down before it can clean up all it's resources (Redis/Postgres/Blobstorage/etc.) and these resources are then left behind in integreatly namespace.

This ensures all products are correctly cleaned up before the cloud resource operator is removed. 

### Verification 
- Run the integreatly operator from this branch:
```sh
make cluster/prepare/local
make code/run/service_account
```
- Create an installation cr and wait for the the products to come up
- Delete the installation 
- Ensure that all cro resources (Redis/Postgres/Blobstorage/SMTPCredentialSet) are correctly cleaned up before the `integreatly-cloud-resources` namespace is removed
